File: blogermenia/blog/urls.py
from django.urls import path
from .views.home_views import HomeView, ContactView
from .views.blog_views import (
    BlogListView, BlogDetailView, BlogCreateView,
    BlogUpdateView, BlogDeleteView, BlogLikeView, BlogSaveView,
    BlogShareLinkedInView
)
from .views.playlist_views import (
    PlaylistListView, PlaylistDetailView, PlaylistCreateView,
    PlaylistUpdateView, PlaylistDeleteView,
)
from .views.profile_views import UserProfileView, UserListView, ProfileUpdateView
from .views.pdf_views import GeneratePDFView, CheckPDFStatusView, DownloadPDFView

urlpatterns = [
    # Home
    path('', HomeView.as_view(), name='home'),
    path('contact/', ContactView.as_view(), name='contact'),

    # Blog URLs
    path('blogs/', BlogListView.as_view(), name='blog_list'),
    path('blogs/create/', BlogCreateView.as_view(), name='blog_create'),
    path('blogs/<slug:slug>/', BlogDetailView.as_view(), name='blog_detail'),
    path('blogs/<slug:slug>/update/', BlogUpdateView.as_view(), name='blog_update'),
    path('blogs/<slug:slug>/delete/', BlogDeleteView.as_view(), name='blog_delete'),
    path('blogs/<slug:slug>/like/', BlogLikeView.as_view(), name='blog_like'),
    path('blogs/<slug:slug>/save/', BlogSaveView.as_view(), name='blog_save'),
    path('blogs/<slug:slug>/share-linkedin/', BlogShareLinkedInView.as_view(), name='blog_share_linkedin'),
    
    # PDF URLs
    path('blogs/<slug:slug>/pdf-generate/', GeneratePDFView.as_view(), name='blog_pdf_generate'),
    path('blogs/pdf-status/<str:task_id>/', CheckPDFStatusView.as_view(), name='blog_pdf_status'),
    path('blogs/pdf-download/<str:task_id>/', DownloadPDFView.as_view(), name='blog_pdf_download'),

    # Playlist URLs
    path('playlists/', PlaylistListView.as_view(), name='playlist_list'),
    path('playlists/create/', PlaylistCreateView.as_view(), name='playlist_create'),
    path('playlists/<slug:slug>/', PlaylistDetailView.as_view(), name='playlist_detail'),
    path('playlists/<slug:slug>/update/', PlaylistUpdateView.as_view(), name='playlist_update'),
    path('playlists/<slug:slug>/delete/', PlaylistDeleteView.as_view(), name='playlist_delete'),

    # Profile URLs
    path('accounts-list/', UserListView.as_view(), name='user_list'),
    path('profile/<str:username>/', UserProfileView.as_view(), name='user_profile'),
    path('profile/<str:username>/edit/', ProfileUpdateView.as_view(), name='profile_edit'),
]

File: frontend/lib/urls.ts
/**
 * Every `{% url %}` name in the Django templates, mapped to its Next.js path.
 *
 * Paths keep the trailing slash Django uses (see `trailingSlash` in
 * `next.config.ts`) so links match the original app character for character.
 */
export const urls = {
  home: () => "/",
  contact: () => "/contact/",

  blogList: () => "/blogs/",
  blogListByCategory: (slug: string) => `/blogs/?category=${encodeURIComponent(slug)}`,
  blogListByTag: (tag: string) => `/blogs/?tag=${encodeURIComponent(tag)}`,
  blogCreate: () => "/blogs/create/",
  blogDetail: (slug: string) => `/blogs/${slug}/`,
  blogUpdate: (slug: string) => `/blogs/${slug}/update/`,
  blogDelete: (slug: string) => `/blogs/${slug}/delete/`,
  blogPdf: (slug: string) => `/blogs/${slug}/pdf/`,

  playlistList: () => "/playlists/",
  playlistCreate: () => "/playlists/create/",
  playlistDetail: (slug: string) => `/playlists/${slug}/`,
  playlistUpdate: (slug: string) => `/playlists/${slug}/update/`,
  playlistDelete: (slug: string) => `/playlists/${slug}/delete/`,

  search: (query?: string) => (query ? `/search/?q=${encodeURIComponent(query)}` : "/search/"),

  userList: () => "/accounts-list/",
  userProfile: (username: string) => `/profile/${username}/`,
  profileEdit: (username: string) => `/profile/${username}/edit/`,

  accountLogin: () => "/accounts/login/",
  accountSignup: () => "/accounts/signup/",
  accountLogout: () => "/accounts/logout/",
  accountInactive: () => "/accounts/inactive/",
  accountEmail: () => "/accounts/email/",
  accountChangePassword: () => "/accounts/password/change/",
  accountResetPassword: () => "/accounts/password/reset/",
  accountResetPasswordDone: () => "/accounts/password/reset/done/",
  accountResetPasswordFromKey: (key: string) => `/accounts/password/reset/key/${key}/`,
  accountResetPasswordFromKeyDone: () => "/accounts/password/reset/key/done/",
  accountEmailVerificationSent: () => "/accounts/confirm-email/",
  accountConfirmEmail: (key: string) => `/accounts/confirm-email/${key}/`,

  socialaccountConnections: () => "/accounts/social/connections/",
  socialaccountSignup: () => "/accounts/social/signup/",
  linkedinLogin: () => "/accounts/linkedin_oauth2/login/",
} as const;
